ui/lib/apps/TopSQL/pages/List/ListChart.tsx

```diff
@@ -0,0 +1,182 @@
+
+  useEffect(() => {
+    setDataWithTimeWindowSize({ data, timeWindowSize })
+  }, [data])
+
+  return (
+    <Chart>
+      <Settings
+        theme={theme}
+        showLegend
+        legendPosition={Position.Bottom}
+        onBrushEnd={onBrushEnd}
+        xDomain={{
+          minInterval: dataWithTimeWindowSize.timeWindowSize * 1000,
+          min: timeRangeTimestamp[0] * 1000,
+          max: timeRangeTimestamp[1] * 1000,
+        }}
+      />
+      <Axis
+        id="bottom"
+        position={Position.Bottom}
+        showOverlappingTicks
+        tickFormat={timeFormatter('MM-DD HH:mm:ss')}
+      />
+      <Axis
+        id="left"
+        position={Position.Left}
+        tickFormat={(v) => getValueFormat('ms')(v, 2)}
+      />
+      <BarSeries
+        key="PLACEHOLDER"
+        id="PLACEHOLDER"
+        xScaleType={ScaleType.Time}
+        yScaleType={ScaleType.Linear}
+        xAccessor={0}
+        yAccessors={[1]}
+        stackAccessors={[0]}
+        data={[timeRangeTimestamp[0], 0]}
+        name="PLACEHOLDER"
+      />
+      {Object.keys(chartData).map((digest) => {
+        return (
+          <BarSeries
+            key={digest}
+            id={digest}
+            xScaleType={ScaleType.Time}
+            yScaleType={ScaleType.Linear}
+            xAccessor={0}
+            yAccessors={[1]}
+            stackAccessors={[0]}
+            data={chartData[digest]}
+            name={digestMap?.[digest]?.slice(0, 50) || t('topsql.table.others')}
+          />
+        )
+      })}
+    </Chart>
+  )
+}
+
+function useDigestMap(seriesData: TopsqlSummaryItem[]) {
+  const digestMap = useMemo(() => {
+    if (!seriesData) {
+      return {}
+    }
+    return seriesData.reduce((prev, { sql_digest, sql_text }) => {
+      prev[sql_digest!] = sql_text
+      return prev
+    }, {})
+  }, [seriesData])
+  return { digestMap }
+}
+
+function useChartData(seriesData: TopsqlSummaryItem[]) {
+  const chartData = useMemo(() => {
+    if (!seriesData) {
+      return {}
+    }
+    // Group by SQL digest + timestamp and sum their values
+    const valuesByDigestAndTs: Record<string, Record<number, number>> = {}
+    const sumValueByDigest: Record<string, number> = {}
+    seriesData.forEach((series) => {
+      const seriesDigest = series.sql_digest!
+
+      if (!valuesByDigestAndTs[seriesDigest]) {
+        valuesByDigestAndTs[seriesDigest] = {}
+      }
+      const map = valuesByDigestAndTs[seriesDigest]
+      let sum = 0
+      series.plans?.forEach((values) => {
+        values.timestamp_sec?.forEach((t, i) => {
+          if (!map[t]) {
+            map[t] = values.cpu_time_ms![i]
+          } else {
+            map[t] += values.cpu_time_ms![i]
+          }
+          sum += values.cpu_time_ms![i]
+        })
+      })
+
+      if (!sumValueByDigest[seriesDigest]) {
+        sumValueByDigest[seriesDigest] = 0
+      }
+      sumValueByDigest[seriesDigest] += sum
+    })
+
+    // Order by digest
```

Comment:
I'm not sure, but may be you will want to order by the sum CPU time, in order to make the chart looks better. (this is how https://github.com/breeswish/top-sql-playground does)